core/src/test/java/io/jenkins/pluginhealth/scoring/probes/NumberOfOpenIssuesProbeTest.java

```diff
@@ -0,0 +1,235 @@
+        when(ctx.getRepositoryName(scmLink)).thenReturn(Optional.of(repository));
+        when(gh.getRepository(repository)).thenReturn(ghRepository);
+        when(ghRepository.getOpenIssueCount()).thenReturn(10);
+
+        final NumberOfOpenIssuesProbe probe = getSpy();
+
+        assertThat(probe.apply(plugin, ctx))
+            .usingRecursiveComparison()
+            .comparingOnlyFields("id", "status", "message")
+            .isEqualTo(ProbeResult.success(NumberOfOpenIssuesProbe.KEY, "6 open issues found in JIRA. 10 open issues found in GitHub."));
+        verify(probe, atMostOnce()).doApply(plugin, ctx);
+    }
+
+    @Test
+    void shouldBeAbleToFindNumberOfOpenIssuesOnlyInJira() throws JsonProcessingException {
+        final String pluginName = "mailer";
+        final String repository = "jenkinsci/" + pluginName + "-plugin";
+        final String scmLink = "https://github.com/" + repository;
+
+        final Plugin plugin = mock(Plugin.class);
+        final ProbeContext ctx = mock(ProbeContext.class);
+        JsonNode jsonNode = mock(JsonNode.class);
+        RestTemplate restTemplate = mock(RestTemplate.class);
+        ResponseEntity responseEntity = mock(ResponseEntity.class);
+
+        final String jsonString = "{\"expand\":\"names,schema\",\"startAt\":0,\"maxResults\":50,\"total\":1,\"issues\":[]}";
+
+        when(plugin.getDetails()).thenReturn(
+            Map.of(
+                SCMLinkValidationProbe.KEY, ProbeResult.success(SCMLinkValidationProbe.KEY, ""),
+                UpdateCenterPluginPublicationProbe.KEY, ProbeResult.success(UpdateCenterPluginPublicationProbe.KEY, "")
+            )
+        );
+
+        when(ctx.getUpdateCenter()).thenReturn(new UpdateCenter(
+            Map.of(),
+            Map.of(),
+            List.of(),
+            List.of(
+                Map.of(
+                    "reportUrl",
+                    "https://www.jenkins.io/participate/report-issue/redirect/#18331",
+                    "type",
+                    "jira",
+                    "viewUrl",
+                    "https://issues.jenkins.io/issues/?jql=component=18331"
+                )
+            )
+        ));
+
+        when(plugin.getName()).thenReturn(pluginName);
+        when(plugin.getScm()).thenReturn(scmLink);
+        when(ctx.getRepositoryName(scmLink)).thenReturn(Optional.of(repository));
+        when(restTemplate.getForEntity(anyString(), anyString().getClass())).thenReturn(responseEntity);
+        when(responseEntity.getBody()).thenReturn(jsonString);
+        when(jsonNode.get(anyString())).thenReturn(new ObjectMapper().readTree(jsonString));
+
+        final NumberOfOpenIssuesProbe probe = getSpy();
+
+        assertThat(probe.apply(plugin, ctx))
+            .usingRecursiveComparison()
+            .comparingOnlyFields("id", "status", "message")
+            .isEqualTo(ProbeResult.success(NumberOfOpenIssuesProbe.KEY, "67 open issues found in JIRA."));
```

Comment:
Since this is a live plugin. I wanted to mock the total number `67`.

But I wasn't able to to. From the debugger I could make a guess the API call in the code overrides the mock. But am not sure.